lib/src/workspace.rs

```diff
@@ -150,24 +151,28 @@ fn init_working_copy(
 impl Workspace {
     pub fn new(
         workspace_root: &Path,
+        repo_path: PathBuf,
         working_copy: Box<dyn WorkingCopy>,
         repo_loader: RepoLoader,
     ) -> Result<Workspace, PathError> {
         let workspace_root = workspace_root.canonicalize().context(workspace_root)?;
         Ok(Self::new_no_canonicalize(
             workspace_root,
+            repo_path,
```

Comment:
Maybe it's safer to do `repo_path.canonicalize()` here? It was previously obtained from `repo.loader()`, which I think was canonicalized.

Reply:
Oh, but `load()`-ed workspace wasn't? That's odd.